optakt/optakt.github.io
# Testing guide page

# Goal of this PR

Fixes #7 
Fixes #4 
Fixes #3 

This PR also adds an MIT license to the repository, and a simple description of what this repo is about in the README, as well as a number of features to our mkdocs setup:

* Admonitions, which enable us to write notes, examples, warnings etc. that can be folded or expanded.
* Markdown includes, which keep our documentation tidy by including external files into our compiled markdown before it is rendered into a website.
* MkDocs excludes, which prevents MkDocs from trying to compile the files we want to include.
* Automatically detects whether machine uses dark or light theme and adjusts the palette of the docs to it. Also adds a button at the top right of the site to manually toggle between dark/light.

## Examples

This is a warning admonition that is expanded by default.

![SharedScreenshot](https://user-images.githubusercontent.com/6976628/141313356-9aa38759-a54c-4dee-a0b1-78641f7facf4.jpg)

These are example admonitions which contain code examples and are collapsed by default. Users can expand them to read the code, and fold them back.

![SharedScreenshot](https://user-images.githubusercontent.com/6976628/141313509-a445aec1-0ae1-45e0-bed2-8fe78bb92e1b.jpg)

## Additional Notes

I will explain how to use admonitions and imports for this repository when implementing the section about documentation guidelines in issue #6 

## How to test it

Dependencies: Python3, Pip.

* Check out this branch on your machine
* `pip install -r requirements.txt`
* `mkdocs build`
* Open `site/testing/index.html` in whatever browser.
